File: passgen-app/passgen-web/src/main/java/com/passgen/core/actors/GeneratePasswordActor.java
package com.passgen.core.actors;

import akka.actor.ActorRef;
import akka.actor.UntypedActor;
import akka.event.Logging;
import akka.event.LoggingAdapter;
import com.passgen.core.model.GeneratePassword;
import org.apache.commons.lang.RandomStringUtils;

import java.security.SecureRandom;

import static com.passgen.core.model.PasswordResult.withPassword;

public class GeneratePasswordActor extends UntypedActor {
    private final LoggingAdapter LOG = Logging.getLogger(getContext().system(), this);
    public static final int DEFAULT_LENGTH = 8;

    private final ActorRef next;

    public GeneratePasswordActor(ActorRef next) {
        this.next = next;
    }

    @Override
    public void onReceive(Object m) throws Exception {
        if (m instanceof GeneratePassword) {
            LOG.info("Generate password: {}", m);
            GeneratePassword cmd = (GeneratePassword)m;
            int size = cmd.length > 0 ? cmd.length : DEFAULT_LENGTH;
            String pass = RandomStringUtils.random(size, 0, 0, true, true, null, new SecureRandom());
            LOG.info("Password: {}", pass);
            next.tell(withPassword(pass), getSender());
        } else {
            unhandled(m);
        }
    }
}

File: passgen-app/passgen-scala/src/main/scala/com/passgen/core/actors/GeneratePasswordActor.scala
package com.passgen.core.actors

import java.security.SecureRandom

import akka.actor._
import akka.event.slf4j.SLF4JLogging
import com.passgen.core.model.GeneratePassword
import com.passgen.core.model.PasswordResult._
import org.apache.commons.lang.RandomStringUtils

class GeneratePasswordActor(next: ActorRef) extends Actor with SLF4JLogging {
  val DEFAULT_LENGTH: Int = 8

  def receive = {
    case m: GeneratePassword =>
      log.debug("Generate password: {}", m)
      val size: Int = if (m.length > 0) m.length else DEFAULT_LENGTH
      val pass: String = RandomStringUtils.random(size, 0, 0, true, true, null, new SecureRandom)
      log.debug("Password: {}", pass)
      next.tell(withPassword(pass), sender())
  }
}